packages/opencensus-web-instrumentation-zone/src/interaction-tracker.ts

```diff
@@ -29,39 +31,93 @@ export type AsyncTask = Task & {
 };
 
 export class InteractionTracker {
+  private readonly tracingZones: { [index: string]: Zone } = {};
   constructor() {
+    const interactionTracker: InteractionTracker = this;
+
     const runTask = Zone.prototype.runTask;
     Zone.prototype.runTask = function(
       task: AsyncTask,
       applyThis: unknown,
       applyArgs: unknown
     ) {
-      if (task.eventName && task.eventName.toString().indexOf('click') !== -1) {
-        console.log('Running task');
+      const time = Date.now();
+
+      console.warn('Running task');
+      console.log(task.zone);
+
+      let taskZone = this;
+      if (isTrackedElement(task)) {
         console.log('Click detected');
+
+        const zoneName = randomSpanId();
+        const tracingZone = Zone.root.fork({
+          name: zoneName,
+          properties: {
+            isTracingZone: true,
+            tracingId: randomTraceId(),
+          },
+        });
+
+        interactionTracker.tracingZones[zoneName] = tracingZone;
+
+        // Change the zone task.
+        // tslint:disable:no-any
+        (task as any)._zone = tracingZone;
+        taskZone = tracingZone;
+        console.log('New zone:');
+        console.log(taskZone);
+      } else {
+        // If we already are in a tracing zone, just run the task in our tracing zone.
+        if (task.zone && interactionTracker.tracingZones[task.zone.name]) {
```

Comment:
Could you just check the `isTracingZone` property on the Zone data, which would then make it so you don't need to store the object of `tracingZones`? Or is that not easily accessible?